libmachine/provision/provisiontest/sshcommander.go

```diff
@@ -0,0 +1,22 @@
+//Package provisiontest provides utilities for testing provisioners
+package provisiontest
+
+import (
+	"errors"
+	"strings"
+)
+
+//FakeSSHCommander is an implementation of provision.SSHCommander to provide predictable responses set by testing code
+//Extend it when needed
+type FakeSSHCommander struct {
+	//Result of the ssh command to look up the FilesystemType
+	FilesystemType string
```

Comment:
This would indeed be perfect for testing the output parsing.
For testing more high level functionality, like checking the default graphdriver in the RedHat provisioner for example, it does not really make sense to know that the underlying call is `stat -f -c %T /var/lib`, this would also prevent easy refactoring of the specific implementation of the filesystem checking without modifying all tests.
Both kind of tests are necessary so I'll refactor the FakeSSHCommander to use an internal command/resonse map for output parsing tests while a creationmethod with more high level configuration can be used by the other ones.